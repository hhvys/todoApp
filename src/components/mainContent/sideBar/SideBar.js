import React from 'react';
import SearchBar from './searchBar/SearchBar';
import TabList from './tabList/TabList';
import CreateList from './createList/CreateList';
import './sideBar.css';

const SideBar = () => {
	return (
		<div className="side-bar fixed">
			<SearchBar className="search-bar"/>
			<TabList className="tab-list"/>
			<CreateList className="create-list"/>
		</div>
	);
};

export default SideBar;