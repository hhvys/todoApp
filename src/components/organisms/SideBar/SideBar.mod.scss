@import "../../../constants.mod";

.sideBar {
  width: $side-bar-width+px;
}

.searchBar {
  height: 45px;
  min-height: 45px;
  background-color: #5B795A
}

.verticalTabContainer {
  background-color: $white-background;
  width: $side-bar-width+px;
  overflow-y: auto;
  overflow-x: hidden;
  flex: 1
}

.verticalTabContainer::-webkit-scrollbar {
  width: 5px;
}

.verticalTabContainer::-webkit-scrollbar-track {
  background-color: #f1f1f1;
}

.verticalTabContainer::-webkit-scrollbar-thumb {
  background-color: darkgrey;
  outline: 1px solid slategrey;
  border-radius: 5px;
}

.footer {
  height: 42px;
  min-height: 42px;
  background-color: $white-background;
  border-top: 1px solid $border-color;
  color: $side-bar-primary;
  font-weight: 500;
  fill: $side-bar-primary;
}

.footerLogo {
  fill: $side-bar-primary;
  width: 30px;
}
