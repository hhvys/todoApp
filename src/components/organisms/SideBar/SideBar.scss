.tab {
  height: 42px;
  min-height: 42px;
  background-color: #F7F7F7;
  border-top: 1px solid #e0e0df;
  color: #328ad6;
  font-weight: 500;
  fill: #328ad6
}
