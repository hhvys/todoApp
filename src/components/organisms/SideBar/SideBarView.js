import React from 'react';
import SearchBar from "../../molecules/SearchBar/SearchBar";
import VerticalTab from "../../molecules/VerticalTab/VerticalTab";
import {PLUS, LIST_SYMBOL, PENCIL} from '../../atoms/logos/constants';
import ListContainer from '../../../containers/ListContainer';
import CollapsedSideBar from "../CollapsedSideBar/CollapsedSideBar";
import styles from './SideBar.mod.scss';


class SideBarView extends React.Component {

	render() {
		const {
			className,
			onSearchChange,
			onCreateClick,
			collapseSideBar,
			collapsed,
			searchValue,
			...props
		} = this.props;

		return collapsed ?
			<CollapsedSideBar collapseSideBar={collapseSideBar}/> :
			(
				<div
					className={`${styles.sideBar} ${className ? className : ''} d-flex flex-column align-items-center justify-content-center`}
					{...props}>
					<SearchBar
						className={styles.searchBar}
						onInputChange={onSearchChange}
						collapseSideBar={collapseSideBar}
						inputValue={searchValue}
					/>

					<ListContainer
						className={styles.verticalTabContainer}
						headerSymbol={LIST_SYMBOL}
						footerSymbol={PENCIL}
						/>

					<VerticalTab headerSymbol={PLUS}
											 mainContent={"Create List"}
											 onClick={onCreateClick}
											 headerClass={styles.footerLogo}
											 className={styles.footer}
					/>
				</div>
			);
	}
}

export default SideBarView;
