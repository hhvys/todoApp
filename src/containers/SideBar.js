import {connect} from 'react-redux';
import SideBarView from '../components/organisms/SideBar/SideBarView';
import {searchQuery, toggleModal, toggleSideBar} from "../actions/actionCreaters";
import {getTabs} from "../reducers/tabs/tabs";
import {getActiveTab} from "../reducers/activeTab";
import {getCollapsedSideBar} from "../reducers/collapsedSideBar";

const mapStateToProps = (state) => {

	return {
		activeTab: getActiveTab(state),
		tabs: getTabs(state),
		collapsed: getCollapsedSideBar(state)
	};

};

const mapDispatchToProps = (dispatch) => {
	return {
		onSearchChange: (query) => dispatch(searchQuery(query)),
		onCreateClick: () => dispatch(toggleModal()),
		collapseSideBar: () => dispatch(toggleSideBar())
	};
};

const SideBar = connect(
	mapStateToProps,
	mapDispatchToProps
)(SideBarView);

export default SideBar;

