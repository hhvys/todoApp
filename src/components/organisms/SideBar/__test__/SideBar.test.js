import React from 'react';
import SideBar from '../SideBarView';
jest.mock('node-uuid', () => ({v4: jest.fn(() => 1)}));

describe('SideBar', () => {
	let props;
	let mountedSideBar;

	const sideBar = () => {
		if (!mountedSideBar) {
			mountedSideBar = shallow(
				<SideBar {...props}/>
			);
		}
		return mountedSideBar;
	};

	beforeEach(() => {
		props = {
			className: "className",
			onSearchChange: jest.fn(),
			onCreateClickonCreateClick: jest.fn(),
			collapseSideBar: jest.fn(),
			collapsed: false,
			searchValue: ''
		};
		mountedSideBar = undefined;
	});

	it('should render CollapsedSideBar if collapsed is true', () => {
		props.collapsed = true;
		const collapsedSideBar = sideBar().find('CollapsedSideBar');
		expect(collapsedSideBar.length).toBe(1);
	});
	it('should render ListContainerView if collapsed is false', () => {
		props.collapsed = false;
		const listContainer = sideBar().find('Connect(ListContainerView)');
		expect(listContainer.length).toBe(1);
	});
	it('should render SearchBar if collapsed is false', () => {
		props.collapsed = false;
		const searchBar = sideBar().find('SearchBar');
		expect(searchBar.length).toBe(1);
	});
	it('should render RowComponent if collapsed is false', () => {
		props.collapsed = false;
		const rowComponent = sideBar().find('RowComponent');
		expect(rowComponent.length).toBe(1);
	});
	it('should pass searchValue to inputValue of SearchBar', () => {
		props.searchValue = 'searchValue';
		const searchBar = sideBar().find('SearchBar');
		expect(searchBar.props().inputValue).toBe(props.searchValue);
	});

});
