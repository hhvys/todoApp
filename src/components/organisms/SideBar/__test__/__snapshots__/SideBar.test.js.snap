// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render CollapsedSideBar if collapsed is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarView
    className="className"
    collapseSideBar={[Function]}
    collapsed={true}
    onCreateClickonCreateClick={[Function]}
    onSearchChange={[Function]}
    searchValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "collapseSideBar": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "collapseSideBar": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SideBar should render ListContainerView if collapsed is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarView
    className="className"
    collapseSideBar={[Function]}
    collapsed={false}
    onCreateClickonCreateClick={[Function]}
    onSearchChange={[Function]}
    searchValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBar
          className="searchBar"
          collapseSideBar={[Function]}
          inputValue=""
          onInputChange={[Function]}
/>,
        <Connect(ListContainerView)
          className="verticalTabContainer"
          footerIcon="PENCIL"
          headerIcon="LIST_ICON"
/>,
        <RowComponent
          className="footer"
          headerClass="footerLogo"
          headerIcon="PLUS"
          mainContent="Create List"
          onClick={undefined}
/>,
      ],
      "className": "sideBar className d-flex flex-column align-items-center justify-content-center",
      "onCreateClickonCreateClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "searchBar",
          "collapseSideBar": [Function],
          "inputValue": "",
          "onInputChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "verticalTabContainer",
          "footerIcon": "PENCIL",
          "headerIcon": "LIST_ICON",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "footer",
          "headerClass": "footerLogo",
          "headerIcon": "PLUS",
          "mainContent": "Create List",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBar
            className="searchBar"
            collapseSideBar={[Function]}
            inputValue=""
            onInputChange={[Function]}
/>,
          <Connect(ListContainerView)
            className="verticalTabContainer"
            footerIcon="PENCIL"
            headerIcon="LIST_ICON"
/>,
          <RowComponent
            className="footer"
            headerClass="footerLogo"
            headerIcon="PLUS"
            mainContent="Create List"
            onClick={undefined}
/>,
        ],
        "className": "sideBar className d-flex flex-column align-items-center justify-content-center",
        "onCreateClickonCreateClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "searchBar",
            "collapseSideBar": [Function],
            "inputValue": "",
            "onInputChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "verticalTabContainer",
            "footerIcon": "PENCIL",
            "headerIcon": "LIST_ICON",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "footer",
            "headerClass": "footerLogo",
            "headerIcon": "PLUS",
            "mainContent": "Create List",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SideBar should render RowComponent if collapsed is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarView
    className="className"
    collapseSideBar={[Function]}
    collapsed={false}
    onCreateClickonCreateClick={[Function]}
    onSearchChange={[Function]}
    searchValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBar
          className="searchBar"
          collapseSideBar={[Function]}
          inputValue=""
          onInputChange={[Function]}
/>,
        <Connect(ListContainerView)
          className="verticalTabContainer"
          footerIcon="PENCIL"
          headerIcon="LIST_ICON"
/>,
        <RowComponent
          className="footer"
          headerClass="footerLogo"
          headerIcon="PLUS"
          mainContent="Create List"
          onClick={undefined}
/>,
      ],
      "className": "sideBar className d-flex flex-column align-items-center justify-content-center",
      "onCreateClickonCreateClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "searchBar",
          "collapseSideBar": [Function],
          "inputValue": "",
          "onInputChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "verticalTabContainer",
          "footerIcon": "PENCIL",
          "headerIcon": "LIST_ICON",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "footer",
          "headerClass": "footerLogo",
          "headerIcon": "PLUS",
          "mainContent": "Create List",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBar
            className="searchBar"
            collapseSideBar={[Function]}
            inputValue=""
            onInputChange={[Function]}
/>,
          <Connect(ListContainerView)
            className="verticalTabContainer"
            footerIcon="PENCIL"
            headerIcon="LIST_ICON"
/>,
          <RowComponent
            className="footer"
            headerClass="footerLogo"
            headerIcon="PLUS"
            mainContent="Create List"
            onClick={undefined}
/>,
        ],
        "className": "sideBar className d-flex flex-column align-items-center justify-content-center",
        "onCreateClickonCreateClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "searchBar",
            "collapseSideBar": [Function],
            "inputValue": "",
            "onInputChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "verticalTabContainer",
            "footerIcon": "PENCIL",
            "headerIcon": "LIST_ICON",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "footer",
            "headerClass": "footerLogo",
            "headerIcon": "PLUS",
            "mainContent": "Create List",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SideBar should render SearchBar if collapsed is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarView
    className="className"
    collapseSideBar={[Function]}
    collapsed={false}
    onCreateClickonCreateClick={[Function]}
    onSearchChange={[Function]}
    searchValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBar
          className="searchBar"
          collapseSideBar={[Function]}
          inputValue=""
          onInputChange={[Function]}
/>,
        <Connect(ListContainerView)
          className="verticalTabContainer"
          footerIcon="PENCIL"
          headerIcon="LIST_ICON"
/>,
        <RowComponent
          className="footer"
          headerClass="footerLogo"
          headerIcon="PLUS"
          mainContent="Create List"
          onClick={undefined}
/>,
      ],
      "className": "sideBar className d-flex flex-column align-items-center justify-content-center",
      "onCreateClickonCreateClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "searchBar",
          "collapseSideBar": [Function],
          "inputValue": "",
          "onInputChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "verticalTabContainer",
          "footerIcon": "PENCIL",
          "headerIcon": "LIST_ICON",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "footer",
          "headerClass": "footerLogo",
          "headerIcon": "PLUS",
          "mainContent": "Create List",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBar
            className="searchBar"
            collapseSideBar={[Function]}
            inputValue=""
            onInputChange={[Function]}
/>,
          <Connect(ListContainerView)
            className="verticalTabContainer"
            footerIcon="PENCIL"
            headerIcon="LIST_ICON"
/>,
          <RowComponent
            className="footer"
            headerClass="footerLogo"
            headerIcon="PLUS"
            mainContent="Create List"
            onClick={undefined}
/>,
        ],
        "className": "sideBar className d-flex flex-column align-items-center justify-content-center",
        "onCreateClickonCreateClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "searchBar",
            "collapseSideBar": [Function],
            "inputValue": "",
            "onInputChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "verticalTabContainer",
            "footerIcon": "PENCIL",
            "headerIcon": "LIST_ICON",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "footer",
            "headerClass": "footerLogo",
            "headerIcon": "PLUS",
            "mainContent": "Create List",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
